src/components/filters/Date/index.tsx

```diff
@@ -49,40 +55,43 @@ function DateFilter({ disabled, header, selectableTableStoreName }: Props) {
         },
         setFilter: (option: StatsFilter) => {
             setStatsFilter(option);
-            setCurrentOption(option);
             handlers.closeMenu();
         },
     };
 
+    const optionProps = {
+        currentOption: statsFilter,
```

Comment:
Eventually we really need to move the store name into a context so we don't have to pass stuff from a store.